frontEnd-Services/src/__tests__/Search.test.js

```diff
@@ -0,0 +1,31 @@
+import { fireEvent, render, screen } from "@testing-library/react";
+import APP_CONSTANTS from "../AppConstants";
+import SearchForm from "../components/header/SearchForm";
+import userEvent from "@testing-library/user-event";
+
+describe("Counter Test Validation", () => {
+  it("Test that a click event on decrement button decrements the displayed value", () => {
+    const searchInPutHandler = jest.fn();
+
+    render(
+      <SearchForm
+        defaultSearch="Avengers"
+        searchTextData={searchInPutHandler}
+      ></SearchForm>
+    );
+    let searchButtonTextBox = screen.getByPlaceholderText(
+      /what do you want to watch\?/i
+    );
+    userEvent.type(searchButtonTextBox, "Avengers");
+    expect(searchButtonTextBox).toBeTruthy();
+    expect(searchButtonTextBox.value).toBe("Avengers");
+    userEvent.click(
+      screen.getByRole("button", {
+        name: /search/i,
+      })
+    );
+    expect(searchInPutHandler).toHaveBeenCalled();
+    // userEvent.type(searchButtonTextBox, "Avengers{enter}");
+    // expect(searchButtonTextBox.value).toBe("Avengers");
+  });
+});
```

Comment:
Hi @ravitejamuppalla , as per the test description counter value should be decremented but you are testing some other component called `SearchForm` .

Also, if you are writing test cases for `SearchForm`, not all possible scenarios are covered, like -

- On change of input text what should happened
- If invalid input text is provided what should be shown.
- If user types the input and press `enter` key then what's the outcome.